spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/NodeLocator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.connection;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.amqp.AmqpException;
+import org.springframework.core.log.LogAccessor;
+import org.springframework.lang.Nullable;
+
+/**
+ * Used to obtain a connection factory for the queue leader.
+ * @param <T> the client type.
+ *
+ * @author Gary Russell
+ * @since 2.4.8
```

Comment:
`3.0`?
Or do you really want a back-port for this change?

Reply:
The interfaces were only moved, so they should keep their `@since` - I am not fixing the tangle in 2.4.x.